src/ScrollView.android.tsx

```diff
@@ -16,100 +16,116 @@ export default React.forwardRef(
+    });
+    const enableMvcpWithRetriesRef = useRef(() => {
+      // debounce to wait till consecutive mvcp enabling
+      // this ensures that always previous handles are disabled first
+      if (debounceTimeoutId.current) {
+        clearTimeout(debounceTimeoutId.current);
       }
+      debounceTimeoutId.current = setTimeout(async () => {
+        // disable any previous enabled handles
+        await disableMvcpRef.current();
+
+        if (
+          !flRef.current ||
+          !isMvcpPropPresentRef.current ||
+          isMvcpEnabledNative.current ||
+          Platform.OS !== 'android'
+        ) {
+          return;
+        }
+        const scrollableNode = flRef.current.getScrollableNode();
+
+        try {
+          const _handle: number = await ScrollViewManager.enableMaintainVisibleContentPosition(
+            scrollableNode,
+            autoscrollToTopThreshold.current,
+            minIndexForVisible.current
+          );
+          handle.current = _handle;
+        } catch (error: any) {
+          /**
+           * enableMaintainVisibleContentPosition from native module may throw IllegalViewOperationException,
+           * in case view is not ready yet. In that case, lets do a retry!! (max of 10 tries)
+           */
+          if (enableMvcpRetriesCount.current < 10) {
+            retryTimeoutId.current = setTimeout(
+              enableMvcpWithRetriesRef.current,
+              100
+            );
+            enableMvcpRetriesCount.current += 1;
+          }
+        }
+      }, 300);
+    });
 
-      /**
-       * If the enableMaintainVisibleContentPosition has already been called, then
-       * lets not call it again, unless prop values of mvcp changed.
-       *
-       * This condition is important since `resetMvcpIfNeeded` gets called in refCallback,
-       * which gets called by react on every update to list.
-       */
-      if (isMvcpEnabled.current && !hasMvcpChanged) {
-        return;
+    useEffect(() => {
+      // when the mvcp prop changes
+      // enable natively again, if the prop has changed
+      const propAutoscrollToTopThreshold =
+        props.maintainVisibleContentPosition?.autoscrollToTopThreshold ||
+        -Number.MAX_SAFE_INTEGER;
+      const propMinIndexForVisible =
+        props.maintainVisibleContentPosition?.minIndexForVisible || 1;
+      const hasMvcpChanged =
+        autoscrollToTopThreshold.current !== propAutoscrollToTopThreshold ||
+        minIndexForVisible.current !== propMinIndexForVisible ||
+        isMvcpPropPresentRef.current !== !!props.maintainVisibleContentPosition;
+
+      if (hasMvcpChanged) {
+        enableMvcpRetriesCount.current = 0;
+        autoscrollToTopThreshold.current = propAutoscrollToTopThreshold;
+        minIndexForVisible.current = propMinIndexForVisible;
+        isMvcpPropPresentRef.current = !!props.maintainVisibleContentPosition;
+        enableMvcpWithRetriesRef.current();
       }
+    }, [props.maintainVisibleContentPosition]);
```

Comment:
maybe we should add individual properties of `maintainVisibleContentPosition` here, since its an object. And inline objects might cause un-necessary re-renders.

Reply:
in my view, not needed.. as there is a comparison inside the use effect and if a change was detected the enable function runs. So no rerenders, are caused if there wasn't a change of the properties even though the inline object has changed

Reply:
Alright!!